spark/src/test/java/org/apache/iceberg/spark/data/TestHelpers.java

```diff
@@ -267,11 +267,23 @@ private static void assertEqualsUnsafe(Type type, Object expected, Object actual
     }
 
     switch (type.typeId()) {
-      case BOOLEAN:
       case INTEGER:
+        Assert.assertEquals("Values didn't match", ((Number) expected).intValue(),
+                ((Number) actual).intValue());
+        break;
       case LONG:
+        Assert.assertEquals("Values didn't match", ((Number) expected).longValue(),
+                ((Number) actual).longValue());
+        break;
       case FLOAT:
+        Assert.assertEquals("Values didn't match", ((Number) expected).floatValue(),
+                ((Number) actual).floatValue(), 0.01F);
```

Comment:
We generally don't use `assertEquals` that has a floating point tolerance because Iceberg does not calculate values. The values should match exactly. I think this should use `Float.floatToIntBits` and compare the integer results.